Moodle and Opensis templates

vinhub opened the pull request:
This PR contains templates for deploying Moodle and Opensis in single VM and clustered configurations. Details can be found in individual templates. The templates could also serve as samples for people trying to create templates for single VM and clustered deployments of LAMP apps in general.

azurecla commented:
Hi __@vinhub__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

azurecla commented:
__@vinhub__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, AZPRBOT;

singhkay commented on moodle-cluster-ubuntu/azuredeploy.json:
```diff
@@ -0,0 +1,665 @@
+        "West Europe",
+        "East Asia",
+        "Southeast Asia"
+      ],
```
Need to add more locations as VMs are now available in all regions.

singhkay commented on moodle-cluster-ubuntu/azuredeploy.json:
It's better to make apiVersion a variable for easier migration to newer API versions. Also use the GA API version 2015-06-15

singhkay commented on moodle-cluster-ubuntu/azuredeploy.json:
How does the number of NAT rules here relate to the number of VMs created?

singhkay commented on opensis-cluster-ubuntu/azuredeploy.json:
```diff
@@ -0,0 +1,630 @@
+    "lbName": "[concat(parameters('namePrefix'),'LB')]",
+    "lbDBName": "[concat(parameters('namePrefix'),'DBLB')]",
+    "lbID": "[resourceId('Microsoft.Network/loadBalancers',variables('lbName'))]",
+    "lbDBID": "[resourceId('Microsoft.Network/loadBalancers',variables('lbDBName'))]",
```
I don't see this template creating a LB for DB ?

vinhub replied:
This template is designed to deploy only one database server (and multiple front end servers). So no load balancer for database.

singhkay replied:
Then these Ids that refer lbDB should be removed since they aren't being used

singhkay commented:
@vinhub Left a few comments on the moodle template which will apply to the opensis template as well. Also, the cluster templates seem more interesting scenario rather than the single vm ones as it's easy to get a single VM from cluster template. Unless there is some unique value single vm templates add, I suggest only adding the cluster config ones. Thanks!

vinhub commented on moodle-cluster-ubuntu/azuredeploy.json:
It is one NAT rule per front end VM (with an upper limit of 5). We have explained the mapping in the readme:  https://github.com/vinhub/azure-quickstart-templates/tree/moodle-opensis-pr/moodle-cluster-ubuntu#front-end-vms

vinhub commented:
Thanks. We will address the issues you have pointed out asap. 

The clustered configuration templates for Moodle as well as OpenSIS deploy at least one front end and one backend server. So even in the smallest configuration, it will not deploy the single VM configuration where everything is on one VM. Also, the single VM configuration satisfies a lot of use cases so we feel that we need both types of templates.

vinhub commented:
@singhkay We have fixed all issues that you had identified.

All templates:
- Added all currently supported resource locations
- Made apiVersion into a variable and updated it to GA version

Moodle cluster template: Fixed an issue caused by using fqdn for database server by mapping it to ip address in the installation shellscript.

singhkay commented on opensis-cluster-ubuntu/metadata.json:
```diff
@@ -0,0 +1,7 @@
+  "description": "This template deploys OpenSIS Community Edition as a LAMP application on Ubuntu. It creates a one or more Ubuntu VM for the front end and a single VM for the backend. It does a silent install of Apache and PHP on the front end VM's and MySQL on the backend VM. Then it deploys OpenSIS Community Edition on the cluster.  After the deployment is successful, you can go to /opensis-ce ...
+  "summary": "Deploy OpenSIS Community Edition cluster on Ubuntu",
+  "githubUsername": "vinhub",
+  "dateUpdated": "2015-11-04"
```
Don't forget to update the dates before the final merge

singhkay commented:
looks good overall. Few updates needed before merging

vinhub commented:
@singhkay Fixed all the issues you identified in your review comments yesterday.

singhkay commented:
lgtm, merged

singhkay commented:
@vinhub Can you update these templates to make sure they are passing in any passwords to the custom script securely as part of protectedSettings? See here how to do this https://github.com/Azure/azure-quickstart-templates/blob/master/lamp-app/azuredeploy.json#L201
